src/app.rs

```diff
@@ -461,6 +468,52 @@ impl App {
         }
     }
 
+    fn validate_root_age(
+        &self,
+        max_root_age_seconds: i64,
+        root_state: &RootItem,
+    ) -> Result<(), ServerError> {
+        let latest_root = self.tree_state.get_latest_tree().get_root();
+        let processed_root = self.tree_state.get_processed_tree().get_root();
+        let mined_root = self.tree_state.get_mined_tree().get_root();
+
+        match root_state.status {
+            Status::Pending if latest_root == root_state.root => return Ok(()),
+            // Processed status is hidden - this should never happen
+            Status::Processed if processed_root == root_state.root => return Ok(()),
+            // Processed status is hidden so it could be either processed or mined
+            Status::Mined if processed_root == root_state.root || mined_root == root_state.root => {
+                return Ok(())
+            }
+            _ => (),
+        }
+
+        // Latest processed root is always valid
+        if root_state.root == processed_root {
+            return Ok(());
+        }
+
+        let max_root_age = Duration::seconds(max_root_age_seconds);
+
+        let mined_at = root_state.mined_valid_as_of;
+        let pending_as_of = root_state.pending_valid_as_of;
+
+        let now = chrono::Utc::now();
+
```

Comment:
Do you need to initialize varibles here? Or can you just simplify to the following:

```rust
let root_age = if root_state.status == Status::Pending {
    chrono::Utc::now() - root_state.pending_valid_as_of;
} else {
    let mined_at = root_state.mined_valid_as_of.ok_or(ServerError::InvalidRoot)?;
    chrono::Utc::now() - mined_at
};
```

Reply:
Fair, but I'll leave the `now` variable.